backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java

```diff
@@ -777,6 +856,95 @@ private JsonNode getAasSubmodelDescriptors(String manufacturerPartId, String man
         return getAasSubmodelDescriptors(manufacturerPartId, manufacturerId, mpr, --retries);
     }
 
+    /**
+     * Requests an EDR for the communication from edc
+     * <p>
+     * The edc already handles the expiry as configured in the provider data plane and refreshs the token before
+     * answering.
+     *
+     * @param transferProcessId to get the EDR for
+     * @return unpersisted EdrDto.
+     */
+    private EdrDto getEdrForTransferProcessId(String transferProcessId) {
```

Comment:
This method uses the edr API instead of a callback. With TX EDC 0.7.0 and Data Plane Signalling the (dynamic) http receiver extension is not shipped anymore. Thus we need to use the callback. That's possible but the EDC team emphasizes to use the EDR api as the future way.

Beside that, checked implementation. We need to put in the api key into the vault and add the alias during transfer. As a result we would need to add the alias to the application.properties and store the data in the vault. This would be more or less a bidirectional dependency. By using the EDR api only the PURIS application needs to know the EDC api secret, which is also better than now (edc currently needs api key of puris edr api during startup).

I'll increase readability of call as discussed + replace the EDR removal Thread by removing the EDR during `terminateTransfer` method, which is called in the finally part.

Performance will be similar, but maintenance efforts are less.

Reply:
Sorry didn't think about the comment when doing the refactoring. Will keep an eye on that in future :)